genCode.py

```diff
@@ -77,10 +77,24 @@ def createStaticClient(name, api, genAsync=False):
 
     lines.append('    classOptions = {')
 
-    copiedOptions = ('baseUrl', 'exchangePrefix')
+    # if serviceName is not set, get it from the baseUrl, exchangePrefix, or manifest name
```

Comment:
It is in `createApiClient`.  I don't think it makes sense in the constructor, though - this is compatibility for handling older API definitions.  In other words, this is deriving default values for options from the API definition, rather than deriving option values from other options.

Reply:
Yes, it's compatiblity code, but the same code is being specified in here in the code generation and again in createApiClient.  We have other initialization code for clients in the constructor, and by moving this initialization code into the constructor, we don't need to have it duplicated in two places.

But if we remove the guessing of service name with a point release to lib-api, almost all of this code goes away, especially if there's a constant apiVersion of 'v1' added to that patch.